### ISSUE TITLE
Console mode displays 1 FPS

### ISSUE THREAD

**Issue by TediusTimmy** [OneLoneCoder/olcPixelGameEngine#298]
Hey Davey,
Whenever I use your new console, my program only reports 1 FPS. I expect that that's just an oversight that someone on the Discord, which I don't follow, has already reported. But, here it is on GitHub, in case other people, like me, don't follow Discord.
![Current Embedded Language](https://user-images.githubusercontent.com/48269399/194743114-6a487822-b0e3-4fa0-81b7-c83c6d0f3d94.png)

**Comment by TediusTimmy** [comment]
Thanks, man. I know that I probably come across as a very confrontational person, but no disrespect (or confrontation, for that matter) is intended. I'm just very blunt. Be blunt to me and I'll tell you if you've crossed a line.

**Comment by OneLoneCoder** [comment]
Can't say I've experienced this myself, I shall investigate.

**Comment by TediusTimmy** [comment]
In this program, I start the console immediately and for it to be running constantly. Maybe that is my problem. Let me create a simple example for you and see if it displays the same properties.

**Comment by TediusTimmy** [comment]
So, this could be a Cygwin/GCC issue: here is code
```
#define OLC_PGE_APPLICATION
#include "olcPixelGameEngine.h"

class Demo : public olc::PixelGameEngine
 {
public:
   Demo()
    {
      sAppName = "Console Test";
    }

private:

public:
   bool OnUserCreate() override
    {
      return true;
    }

   bool OnUserUpdate(float /*fElapsedTime*/) override
    {
      ConsoleShow(olc::Key::ESCAPE);
      return true;
    }

	bool OnConsoleCommand(const std::string& sCommand)
    {
      ConsoleOut() << "> " << sCommand << std::endl;
      return true;
    }
 };

int main()
{
   Demo demo;
   if (demo.Construct(640, 480, 2, 2))
      demo.Start();
   return 0;
}
```
With the Cygwin compile, I get 1 FPS. If you get a different result from MSC++, post a picture and I'll assume it's a Cygwin issue.

**Comment by sigonasr2** [comment]
`ConsoleShow()` by default pauses the timer (Function args are: `ConsoleShow(const olc::Key &keyExit, bool bSuspendTime = true)` . 

Constantly calling it in `OnUserUpdate` will never allow the FPS timer to update, so it remains at 1. You can get a simpler isolated case of this by calling `ConsoleShow(olc::ESCAPE)` in `OnUserCreate()`. This makes the FPS 1 at the beginning of the application. Then, simply hitting Escape to see the FPS timer go back to normal.

Additionally if you set `bSuspendTime` to false, the timer will start updating again, resulting in normal FPS updating.

So for your example @TediusTimmy, doing `ConsoleShow(olc::Key::ESCAPE,false);` would result in an FPS counter updating once again.

**Comment by TediusTimmy** [comment]
@sigonasr2 Something just feels off about this. Yes, the timer is suspended, but it is weird. In my heart of hearts, this is not a bug, but is solved by documentation.

**Comment by TediusTimmy** [comment]
David, i hope the example works for you. Sorry about the Davey, my brother doesn't like being called it either.

**Comment by Moros1138** [comment]
> So, this could be a Cygwin/GCC issue

> With the Cygwin compile, I get 1 FPS. If you get a different result from MSC++, post a picture and I'll assume it's a Cygwin issue.

None of the active testing we do involves cygwin, the current recommendation for using GCC on windows has, and will remain for quite some time due to stability, is GCC provided by MSYS2. If it is an issue specific to cygwin, that's why nobody's run across it.

**Comment by Moros1138** [comment]
that said, I did give your code a quick try on my linux installation and experience a 1fps situation myself.
![2022-10-09_11-46](https://user-images.githubusercontent.com/10980331/194766462-625ddc28-6520-4227-b986-155ebc683804.png)

**Comment by Moros1138** [comment]
Right, after doing a little more research into the console code, it seems there's a second argument for

``ConsoleShow``

``void ConsoleShow(const olc::Key &keyExit, bool bSuspendTime = true);``

which you can see here: <https://github.com/OneLoneCoder/olcPixelGameEngine/blob/master/olcPixelGameEngine.h#L1103>

it defaults to ``true`` which suspends fElapsedTime to 0 each frame... since fElapsedTime is used by the engine to calculate FPS it has the side affect of reporting the FPS inaccurately while time is supsended.

I'll kick this to Javid as I think he'll have a think about whether or not this is simply the intended behavior or if PGE should report FPS accurately whether engine time has halted or not...
